docs/topics/multiplatform/multiplatform-compatibility-guide.md

```diff
@@ -490,3 +490,200 @@ projects.
 **When do the changes take effect?**
 
 In Kotlin 1.9.0, a deprecation warning is introduced when the `android` name is used in Kotlin Multiplatform projects.
+
+<anchor name="declaring-multiple-targets"></anchor>
+## Declaring multiple targets of the same kind
+
+**What's changed?**
+
+We discourage declaring multiple targets of the same kind in a single Gradle project.
+
+Example:
+```kotlin
+kotlin {
+    jvm()
+    jvm("jvmKtor")
+    jvm("jvmOkio") // is not recommended and will produce deprecation warning
+}
+```
+
+**What's the best practice now?**
+
+There are two ways of how to avoid multiple target declaration:
+
+1. Creating different compilations
+
+    If you need to compile some sources with different parameters such as dependencies or compiler options for example
+    for functional tests. Then instead of declaring multiple targets you can create multiple compilations in one target.
+ 
+   <tabs group="build-script">
+   <tab title="Before" group-key="before">
+
+    ```kotlin
+   // Project ':myLibrary:integrationTests'
+   kotlin {
+      jvm("jvmKtor") // !!!
+      jvm("jvmOkio") // !!!
+   
+       sourceSets {
+           // 'commonTest' source set contains shared integration tests of 'myLibrary' between ktor and okio 
+           val commonTest by getting {
+               dependencies {
+                   implementation(projects.myLibrary) 
+               }
+           }
+           // 'jvmKtorTest' source set contains ktor-specific integration tests
+           val jvmKtorTest by getting {
+               dependencies {
+                   implementation(libs.ktor)
+                   implementation(projects.myLibrary.ktor)
+               }
+           }
+   
+           // 'jvmOkioTest' source set contains okio-specific integration tests
+           val jvmOkioTest by getting {
+               dependencies {
+                   implementation(libs.okio)
+                   implementation(projects.myLibrary.okio)
+               }
+           }
+       }
+   }
+    ```
+
+   </tab>
+   <tab title="Now" group-key="now">
+   
+    ```kotlin
+   // Project ':myLibrary:integrationTests'
+   kotlin {
+      jvm {
+         val ktorIntegrationTest by compilations.creating
+         val okioIntegrationTest by compilations.creating
+   
+         // Associate with test compilation to see tests from commonTest
+         ktorIntegrationTest.associateWith(compilations.getByName("test"))
```

Comment:
Why is it associated with `test`? Does it even work in IDE? 

I think the direct counterpart for the previous snippet would be to:

- associate new compilations with `main`
- connect link their default source sets with `dependsOn` to the `commonTest`